src/MahApps.Metro/Styles/Controls.NumericUpDown.xaml

```diff
@@ -0,0 +1,20 @@
+<ResourceDictionary xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+                    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+                    xmlns:Controls="clr-namespace:MahApps.Metro.Controls"
+                    xmlns:Converters="clr-namespace:MahApps.Metro.Converters">
+
+    <ResourceDictionary.MergedDictionaries>
+        <ResourceDictionary Source="pack://application:,,,/MahApps.Metro;component/Styles/Controls.Buttons.xaml" />
+    </ResourceDictionary.MergedDictionaries>
+
+    <Style x:Key="MahApps.Metro.Controls.NumericUpDown.ValueChangedOnFocus"
+           TargetType="{x:Type Controls:NumericUpDown}"
+           BasedOn="{StaticResource {x:Type Controls:NumericUpDown}}">
+        <Style.Resources>
+            <Style x:Key="ChromelessButtonStyle" TargetType="{x:Type ButtonBase}"
+                   BasedOn="{StaticResource ChromelessButtonStyle}">
+                <Setter Property="Focusable" Value="False"/>
+            </Style>
+        </Style.Resources>
+    </Style>
+</ResourceDictionary>
```

Comment:
Why do we need a new style?

Reply:
Yes, we do need a new style. We need TextBox stay in focusing when click increase or decrease button. But we can make this shorter. I committed a new style in my repository.

Reply:
> Yes, we do need a new style. We need TextBox stay in focusing when click increase or decrease button. But we can make this shorter. I committed a new style in my repository.

I think we can get the same result without creating/using a new style. We can prevent focusing of the up and down buttons permanently when we do this

![image](https://user-images.githubusercontent.com/658431/59036599-e8344d80-886f-11e9-9b5d-ca7be95579c4.png)

![image](https://user-images.githubusercontent.com/658431/59036601-eb2f3e00-886f-11e9-8a52-1a5070c84e39.png)

Reply:
Yes, of course. I created a new style just want to keep the original behavior. If you think this is OK. Then we don't need a new style any more.

Reply:
Wait. I think I misunderstood you.

Reply:
I will think about another DepProp, which will control this behavior and check into the PR branch here.

Reply:
The purpose of my style is that when you set `ValueChangedOnLostFocus` to true, then you may want increase button and decrease button only effect the text, not the value. The value only changes when press enter.

Reply:
No, that's not a good UX. The value should changed by using up and down buttons. Only if you type manually we should have this behavior.

Reply:
Yes, but this only happens when you focused on TextBox. If you are not focusing TextBox, then the buttons work.

Reply:
And that's why I created a new style rather than change it directly on the default style. It is just an option.

Reply:
No new style needed. Same can be done with e.g. a DepPropery UpDownButtosFocusable with default true to keep the current implementation. Then another property to handle your new requirement to change the value behind the user input manually by hitting enter.

Reply:
Cool. I am OK with that.